app/lib/features/events/pages/event_list_page.dart

```diff
@@ -74,42 +76,37 @@ class _EventListPageState extends ConsumerState<EventListPage> {
   }
 
   Widget _buildBody() {
-    final calEventsLoader = ref.watch(
-      eventListSearchFilterProvider(
-        (spaceId: widget.spaceId, searchText: searchValue),
-      ),
-    );
-
     return Column(
       crossAxisAlignment: CrossAxisAlignment.stretch,
       children: [
         ActerSearchWidget(
+          initialText: widget.searchQuery,
           onChanged: (value) {
             final notifier = ref.read(searchValueProvider.notifier);
             notifier.state = value;
           },
-          onClear: () => ref.read(searchValueProvider.notifier).state = '',
+          onClear: () {
+            final notifier = ref.read(searchValueProvider.notifier);
+            notifier.state = '';
+          },
+        ),
+        ValueListenableBuilder(
+          valueListenable: eventFilter,
+          builder: (context, showHeader, child) => filterChipsButtons(),
         ),
-        filterChipsButtons(),
         Expanded(
-          child: calEventsLoader.when(
-            data: (calEvents) => _buildEventList(calEvents),
-            error: (error, stack) {
-              _log.severe('Failed to search events in space', error, stack);
-              return ErrorPage(
-                background: const EventListSkeleton(),
-                error: error,
-                stack: stack,
-                onRetryTap: () {
-                  ref.invalidate(
-                    eventListSearchFilterProvider(
-                      (spaceId: widget.spaceId, searchText: searchValue),
-                    ),
-                  );
-                },
-              );
-            },
-            loading: () => const EventListSkeleton(),
+          child: ValueListenableBuilder(
+            valueListenable: eventFilter,
+            builder: (context, showHeader, child) => EventListWidget(
+              spaceId: widget.spaceId,
+              shrinkWrap: false,
+              searchValue: searchValue,
+              eventFiler: eventFilter.value,
```

Comment:
This isn't really the proper riverpod pattern. Passing around a search-value and filter-params and have local valuelisteners instead of just having the proper riverpod state providers and `ref.watch` on them to bubble the information _up_ not _down_ and have all reactively refresh _from_ the state to the tree.